packs/caddy/templates/caddy.nomad.tpl

```diff
@@ -0,0 +1,85 @@
+job [[ template "job_name" . ]] {
+[[ template "region" . ]]
+datacenters = [ [[ range $idx, $dc := .caddy.datacenters ]][[if $idx]],[[end]][[ $dc | quote ]][[ end ]] ]
+  namespace = [[ .caddy.namespace | quote ]]
+
+  type = "system"
+
+  // must have linux for network mode
+  constraint {
+    attribute = "${attr.kernel.name}"
+    value     = "linux"
+  }
```

Comment:
Can we make this into a variable with linux as the default constraint? An example can be seen within the [Prometheus pack](https://github.com/hashicorp/nomad-pack-community-registry/blob/main/packs/prometheus/templates/prometheus.nomad.tpl#L6-L14).